src/azure-cli/azure/cli/command_modules/synapse/manual/operations/workspace.py

```diff
@@ -192,8 +192,14 @@ def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin
                                                                               last_commit_id=last_commit_id,
                                                                               tenant_id=tenant_id)
 
+    if public_network_access is not None:
+        if public_network_access:
+            public_network_access = WorkspacePublicNetworkAccess.ENABLED
+        else:
+            public_network_access = WorkspacePublicNetworkAccess.DISABLED
+
     updated_vnet_settings = ManagedVirtualNetworkSettings(allowed_aad_tenant_ids_for_linking=tenant_ids_list) if allowed_aad_tenant_ids is not None else None
-    workspace_patch_info = WorkspacePatchInfo(tags=tags, sql_administrator_login_password=sql_admin_login_password, encryption=encryption, managed_virtual_network_settings=updated_vnet_settings, workspace_repository_configuration=workspace_repository_configuration, identity=identity)
+    workspace_patch_info = WorkspacePatchInfo(tags=tags, sql_administrator_login_password=sql_admin_login_password, encryption=encryption, managed_virtual_network_settings=updated_vnet_settings, workspace_repository_configuration=workspace_repository_configuration, identity=identity, public_network_access=public_network_access)
```

Comment:
If `public_network_access` is `None`, `None` (but not `WorkspacePublicNetworkAccess.DISABLED`) will be passed along with the API call, is it expected?

Reply:
Yes, if none is in `WorkspacePatchInfo`, then the state is not changed (left as is)